fusion/fusion/FieldExtractor.hpp

```
@@ -0,0 +1,279 @@
+    //! \brief duplicated arguments do not need to be explicitly expressed
+    using OptArg = std::optional<FieldName>;
+
+    //!  \brief compairson function primitives
+    enum FUNC
+    {
+        EQ,
+        LT
+    };
+
+    //!  \brief extract value of named encoding field
+    //!
+    //! handles field name and immediate checking
+    //! \note RS_MAX is overloaded to identify get's for immediate fields
+    uint32_t getField(InstPtrType inst, FieldName field) const
+    {
+        bool isDest = false;
+        if (!checkInstHasField(inst, field, isDest))
+        {
+            throw fusion::FieldExtUnknownField((uint32_t)field,
+                                               inst->dasmString());
+        }
+
+        if (field == FieldName::RS_MAX)
+            return getImmField(inst);
+
+        return getFieldById(inst, field, isDest);
+    }
+
+    //! \brief get the encoded value of a named special field from an
+    //! instptr
+    //!
+    //! This is simpler than getField, a return value of 0 from mavis
+    //! indicates the field does not exist so checkInstHasSField()
+    //! is redundant. getSFieldById() is also not necessary since
+    //! SpecialFields are not distinguished as a source or destination
+    //! operand and can be accessed from the same method.
+    uint32_t getSField(InstPtrType inst, SFieldName field) const
+    {
+        uint32_t value = inst->getSpecialField(field);
+        if (value == 0)
+        {
+            throw fusion::FieldExtUnknownSpecialField((uint32_t)field,
+                                                      inst->dasmString());
+        }
+        return value;
+        //        return getSFieldById(inst, field);
+    }
+
+    //! \brief get the encoded value of the full immediate field
+    //!
+    //! split immediate fields are (will be) ordered msb:lsb and
+    //! concatenated into one unsigned value
+    uint32_t getImmField(InstPtrType inst) const { return 0; }
```

Is this not implemented? If so, should we throw an exception till it is implemented?

---

Yes this is not implemented. I intended to remove this and mark it as a todo.